equinox/_module.py

```diff
@@ -478,11 +480,77 @@ def __new__(
                                     "attempting to override "
                                     f"`{base.__module__}.{base.__qualname__}.{k}`."
                                 )
+
         # [Step 6] Register as a pytree.
+        data_fs = tuple(
+            f.name
+            for f in dataclasses.fields(cls)
+            if not f.metadata.get("static", False)
+        )
+        static_fs = tuple(
+            f.name for f in dataclasses.fields(cls) if f.metadata.get("static", False)
+        )
+
+        def flatten_module(
+            obj: "Module", /
+        ) -> tuple[tuple[PyTree, ...], _FlattenedData]:
+            """Flattens a module into a flattened representation."""
+            d = obj.__dict__
+            sntl = _flatten_sentinel
+
+            # Static fields, filtered on property / init state
+            static_vs = [v for k in static_fs if (v := d.get(k, sntl)) is not sntl]
```

Comment:
I think this will end up being of variable length if we get a sentinel value. That will then produce the wrong behaviour at unflattening time.

We could either similarly filter `static_fs` here, or skip pass sentinels in the unflattening. Wild guess that the latter is more efficient as in general sentinels will be uncommon.

Reply:
Done! I don't see any speedup, but %timeit is an imprecise tool.
This also afforded some further simplifications in `_FlattenedData` and then in `_unflatten_module`